### ISSUE TITLE
Path argument always required.

### ISSUE THREAD

**Issue by lomkju** [RasaHQ/rasa_nlu#954]
**Rasa NLU version** :
Master Branch (Latest) - 0.12.0.aX

**Used backend / pipeline** :
spacy_sklearn

**Operating system** (windows, osx, ...):
Ubuntu 17.10

**Issue**:
Why is the path argument always required ? Making it false in my local seems to work fine . Can you let me know if this is meant to be like this, and why reading it from the config file (yaml) is wrong? 

Latest Branch Code
```
    parser.add_argument('--path',
                        required=False,
                        help="working directory of the server. Models are"
                             "loaded from this directory and trained models "
                             "will be saved here.")
```
Before Merge from config enhancements:

```
    parser.add_argument('--path',
                        required=True,
                        help="working directory of the server. Models are"
                             "loaded from this directory and trained models "
                             "will be saved here.")
```

Best,
Arjun

**Comment by wrathagom** [comment]
If I understand @tmbo's intentions the Config file is just going to be pipeline oriented. All server parameter need to be provided as command line arguments.

**Comment by lomkju** [comment]
@wrathagom @tmbo There seems to be more problem related to emulation -e parameter doesn't seem to work I found that the data_router.py looks for the parameter emulation_mode which is never passed to it from server.py.

data_router.py
```
class DataRouter(object):
    def __init__(self,
                 project_dir=None,
                 max_training_processes=1,
                 response_log=None,
                 emulation_mode=None,
                 remote_storage=None,
                 component_builder=None):

        self._training_processes = max(max_training_processes, 1)
        self.responses = self._create_query_logger(response_log)
        self.project_dir = config.make_path_absolute(project_dir)
        self.emulator = self._create_emulator(emulation_mode)
```

**Comment by alexweidauer** [comment]
looping in @amn41 and @akelad as @tmbo is currently OOO

**Comment by wrathagom** [comment]
sounds like this was closed with #957
